extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1174,11 +1181,31 @@ export class AmpA4A extends AMP.BaseElement {
             'src': srcPath,
             'name': nameData,
           }, SHARED_IFRAME_PROPERTIES));
+      if (method == XORIGIN_MODE.NAMEFRAME) {
+        // TODO(bradfrizzell): change name of function and var
+        const attributes = this.generateSentinelAndContext(iframe);
+        attributes['creative'] = creative;
+        const name = JSON.stringify(attributes);
+        iframe.setAttribute('name', name);
```

Comment:
Wait -- doesn't this overwrite the `name` that we just set in `createElementWithAttributes`?  Seems like you should be able to do this logic in the `case XORIGIN_MODE.NAMEFRAME` block, above, and just let it be inherited in the `createElementWithAttributes` call.